src/ClientData/ScopeStatsCollection.cpp

```diff
@@ -0,0 +1,77 @@
+
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ClientData/ScopeStatsCollection.h"
+
+#include "OrbitBase/Logging.h"
+
+namespace orbit_client_data {
+
+static const ScopeStats kDefaultScopeStats;
+
+ScopeStatsCollection::ScopeStatsCollection(ScopeIdProvider& scope_id_provider,
+                                           const std::vector<const TimerInfo*>& timers) {
+  for (const TimerInfo* timer : timers) {
+    std::optional<ScopeId> scope_id = scope_id_provider.ProvideId(*timer);
+    if (scope_id.has_value()) {
+      UpdateScopeStats(scope_id.value(), *timer);
+    }
+  }
+
+  SortTimers();
+}
+
+void ScopeStatsCollection::UpdateScopeStats(ScopeId scope_id, const TimerInfo& timer) {
+  ScopeStats& stats = scope_stats_[scope_id];
+  const uint64_t elapsed_nanos = timer.end() - timer.start();
+  stats.UpdateStats(elapsed_nanos);
+  scope_id_to_timer_durations_[scope_id].push_back(elapsed_nanos);
+  timer_durations_are_sorted_ = false;
+}
+
+// TODO(b/249046906): Remove this test-only function.
```

Comment:
Thanks! Can you move this comment to the .h file?

Reply:
Done.